docs/retries.md

```diff
@@ -24,14 +24,32 @@ spec:
 
 ## Retry policies
 
-Use `retryPolicy` to choose which failures to retry:
+Use `retryPolicy` to choose which failure types to retry:
 
 - `Always`: Retry all failed steps
-- `OnFailure`: Retry steps whose main container is marked as failed in Kubernetes (this is the default)
+- `OnFailure`: Retry steps whose main container is marked as failed in Kubernetes
 - `OnError`: Retry steps that encounter Argo controller errors, or whose init or wait containers fail
 - `OnTransientError`: Retry steps that encounter errors [defined as transient](https://github.com/argoproj/argo-workflows/blob/master/util/errors/errors.go), or errors matching the `TRANSIENT_ERROR_PATTERN` [environment variable](https://argoproj.github.io/argo-workflows/environment-variables/). Available in version 3.0 and later.
 
-For example:
+The `retryPolicy` applies even if you also specify an `expression`, but in version 3.5 or later the default policy means the expression makes the decision unless you explicitly specify a policy.
+
+The default `retryPolicy` is `OnFailure`, except in version 3.5 or later when an expression is also supplied, when it is `Always`. This may be easier to understand in this diagram.
+
+```mermaid
+flowchart LR
+  start([Start])
+  start --> expression
+  expression(Expression specified)
+  expression-->|No|onfailure
+  expression-->|Yes|version
+  onfailure[retryPolicy: OnFailure]
+  version(Workflows version)
+  version-->|3.4 or ealier|onfailure
+  always[retryPolicy: Always]
+  version-->|3.5 or later|always
+```
+
```

Comment:
Great documentation. I didn't know about mermaid for diagramming. Neat. I know the purpose of the diagram is to show on the right what the "default retryPolicy" is. Another option for even more clarity is to have a branch for whether "retryPolicy" is specified in the Workflow, and for the result to be the net "retryPolicy". But I am also okay the way it is.

Reply:
I've redone this, as I agree it's more helpful. Most people won't care how it works under the hood - they just want to know what will happen when they do whatever it is they do, and hopefully this new chart covers that better.

Reply:
Thank you for doing that! One nitpick is: maybe it would be a little less confusing if the result of `Policy Specified?=No; Expression Specified?=No` were just `OnFailure`?